src/service.rs

```diff
@@ -1440,10 +1957,128 @@ impl Service {
+                                // establishing the connection to the peer via the NAT traversal
+                                // protocol for sending future requests to the peer, if this peer
+                                // was forwarded to us in a NODES response and we hence have a
+                                // relay for it.
+                                if let Some(relay) = active_request.relay {
+                                    if let Some((relay_enr, symmetric_port)) = self.find_enr(&relay)
+                                    {
+                                        if let Some(relay_contact) =
+                                            self.contact_from_enr(&relay_enr, symmetric_port)
+                                        {
+                                            let local_enr = self.local_enr.read().clone();
+                                            if let Some(enr) = active_request.contact.enr() {
+                                                let to_node_id = enr.node_id();
+                                                // Finish one relay request to a given peer before
+                                                // starting another
+                                                if let Entry::Vacant(e) =
+                                                    self.receiver_enrs.entry(to_node_id)
+                                                {
+                                                    e.insert(enr);
+                                                    _ = self.send_relay_request(
+                                                        relay_contact,
+                                                        local_enr,
+                                                        to_node_id,
+                                                    );
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                RequestError::UnusedHolePunchPingToReceiver
+                | RequestError::TimedOutHolePunchPing => {
+                    // Don't update the connection if the request failed as expected according the
+                    // NAT traversal protocol. A RequestError::UnusedHolePunchPingToReceiver means
+                    // that a successful session has already been established. A
+                    // RequestError::TimedOutHolePunchPing usually happens if both nodes are
+                    // behind a NAT and this is the ping from the receiver to the initiator timing
+                    // out but setting the entry in its router's state table for the initiator's
+                    // hole punch ping coming next to enter.
+                    return;
+                }
```

Comment:
`UnusedHolePunchPingToReceiver` looks unreachable to me.

There needs to be an active request to land here, but the `send_hole_punch_ping` fn does not add it if it's hole punch and this errors ir only returned in that case